Code/Framework/AzCore/AzCore/Math/Vector4.inl

```diff
@@ -28,21 +31,41 @@ namespace AZ
         ;
     }
 
-
     AZ_MATH_INLINE Vector4::Vector4(Simd::Vec4::FloatArgType value)
         : m_value(value)
     {
-        ;
     }
 
     AZ_MATH_INLINE Vector4::Vector4(const Vector2& source)
+    : m_value(Simd::Vec4::FromVec2(source.GetSimdValue()))
     {
-        *this = CreateFromVector3(Vector3(source));
+        m_w = 1.0f;
+    }
```

Comment:
Again, I think you need to add an `m_z = 0.0f;` here.

Reply:
> > > We have tried to do this before, in the .inl files but there is a problem here that this causes a circular include.
> > > `Vector4.h includes `Vector4.inl`which is including`Vector3.h`which is including`Vector3.inl`which is includes`Vector4.h
> > 
> > 
> > that's why I've forward declare them in the header and included them in the .inl file it gets around this since the methods are declared ahead of including them when they are used in the .inl file.
> 
> The changes are being built through the AR pipeline right now: https://jenkins.build.o3de.org/job/O3DE/view/change-requests/job/PR-10208/

Well there is a compiliation issue with clang when building on Linux
https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-10208/1/pipeline